pkg/router/gloo.go

```diff
@@ -293,10 +294,20 @@ func (gr *GlooRouter) getGlooUpstreamKubeService(canary *flaggerv1.Canary, svc *
 		Selector:         svc.Spec.Selector,
 	}
 
+	upstreamLabels := make(map[string]string)
+	if upstreamSpec.Labels != nil {
```

Comment:
Why are we making a copy of the labels map?

Reply:
Theoretically because I assumed that there might be default labels passed in at some point. It's a pattern I saw with another operator. Basically, if there is default labels defined somewhere by the operator critical to functionality and a user defines a label with the same key this won't override it. 

This isn't super relevant in this particular case because there are no default labels, but I added it for future compatibility. I can absolutely remove it though.